Src/Support/Google.Apis.Auth/OAuth2/UrlSourcedExternalAccountCredential.cs

```diff
@@ -0,0 +1,127 @@
+                SubjectTokenJsonFieldName = other.SubjectTokenJsonFieldName;
+            }
+        }
+
+        /// <summary>
+        /// The URL from which to obtain the subject token.
+        /// </summary>
+        public string SubjectTokenUrl { get; }
+
+        /// <summary>
+        /// Headers to include in the request for the subject token.
+        /// May be empty. Will not be null.
+        /// </summary>
+        public IReadOnlyDictionary<string, string> Headers { get; }
+
+        /// <summary>
+        /// If set, the subject token response will be parsed as JSON and the
+        /// value in the field with name <see cref="SubjectTokenJsonFieldName"/>
+        /// will be returned as the subject token.
+        /// </summary>
+        public string SubjectTokenJsonFieldName { get; }
+
+        internal UrlSourcedExternalAccountCredential(Initializer initializer) : base(initializer)
+        {
+            SubjectTokenUrl = initializer.SubjectTokenUrl;
+            Headers = initializer.Headers is null
+                ? new ReadOnlyDictionary<string, string>(new Dictionary<string, string>())
+                : new ReadOnlyDictionary<string, string>(initializer.Headers);
+            SubjectTokenJsonFieldName = initializer.SubjectTokenJsonFieldName;
+        }
+
+        /// <inheritdoc/>
+        public override IGoogleCredential MaybeWithScopes(IEnumerable<string> scopes) =>
+            new UrlSourcedExternalAccountCredential(new Initializer(this) { Scopes = scopes });
+
+        /// <inheritdoc/>
+        public override IGoogleCredential WithHttpClientFactory(IHttpClientFactory httpClientFactory) =>
+            new UrlSourcedExternalAccountCredential(new Initializer(this) { HttpClientFactory = httpClientFactory });
+
+        /// <inheritdoc/>
+        public override IGoogleCredential WithQuotaProject(string quotaProject) =>
+            new UrlSourcedExternalAccountCredential(new Initializer(this) { QuotaProject = quotaProject });
+
+        /// <inheritdoc/>
+        protected async override Task<string> GetSubjectTokenAsync(CancellationToken taskCancellationToken)
+        {
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, SubjectTokenUrl);
+            foreach (var headerPair in Headers)
+            {
+                httpRequest.Headers.Add(headerPair.Key, headerPair.Value);
+            }
+
+            var response = await HttpClient.SendAsync(httpRequest, taskCancellationToken).ConfigureAwait(false);
+            var responseText = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+            if (string.IsNullOrEmpty(SubjectTokenJsonFieldName))
+            {
+                return responseText;
+            }
+
+            return NewtonsoftJsonSerializer.Instance.Deserialize<Dictionary<string, string>>(responseText)[SubjectTokenJsonFieldName];
```

Comment:
Are we okay with "leaking" a Json.NET exception if this can't be deserialized? (We probably do already - just thought I'd mention it.)